src/main/java/org/broadinstitute/hellbender/utils/pileup/ReadPileup.java

```diff
@@ -131,6 +159,30 @@ public ReadPileup getPileupForLane(final String laneID) {
     }
 
     /**
+     * Splits the ReadPileup by sample
+     *
+     * @param header              the header to retrieve the samples from
+     * @param assumedSingleSample the sample name if there is no read group
+     * @return a Map of sample name to ReadPileup (including empty pileups for a sample)
+     * @throws org.broadinstitute.hellbender.exceptions.UserException.ReadMissingReadGroup
+     */
+    public Map<String, ReadPileup> splitBySample(final SAMFileHeader header, final String assumedSingleSample) {
+        final Map<String, ReadPileup> toReturn = new HashMap<>();
+        for (final String sample : getSamples(header)) {
+            final ReadPileup pileupBySample = getPileupForSample(sample, header);
```

Comment:
Added support for `null` sample in `getPileupForSample` may solve the NPE issue.